src/app/modules/notification/services/notification.service.ts

```diff
@@ -0,0 +1,111 @@
+import { Injectable } from '@angular/core';
+
+import { Notification } from '../@type/notification';
+import { NotificationOptions } from '../@type/notification-options';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class NotificationService {
+
+  /** Default notifiactions options */
+  options: NotificationOptions = {
+    isDismissable: true,
+    isSelfClosing: true,
+    duration: 5
+  };
+
+  /** Notifications list */
+  private notifications: Notification[] = [];
+
+  /** Id prefix of the nofications */
+  private idPrefix = 'notification-';
+
+  /** Delay to wait before removing a notification on a dismiss action */
+  private removingDelay = 1000;
+
+  constructor() { }
+
+  /**
+   * Get all active notifications
+   */
+  getNotifications(): Notification[] {
+    return this.notifications;
+  }
+
+  /**
+   * Get removing delay
+   */
+  getRemovingDelay(): number {
+    return this.removingDelay;
+  }
+
+  /**
+   * Add a notification
+   */
+  push(message: string, type: string = 'default', options = {}): void {
+    const notification = {
+      id: this.generateUniqueId(),
+      message: message,
+      type: type,
+      options: this.mergeWithDefaultOptions(options)
+    };
+
+    this.notifications.push(notification);
+
+    if (notification.options.isSelfClosing) {
+      this.startTimer(notification);
+    }
+  }
+
+  /**
+   * Remove a notification
+   */
+  dismiss(notification: Notification, delay: number = this.removingDelay): void {
+    const index = this.notifications.indexOf(notification);
+    if (index > -1) {
+      setTimeout(() => {
+        this.notifications.splice(index, 1);
+      }, delay);
+    }
+  }
+
+  /**
+   * Clear all notifications
+   */
+  dismissAll(): void {
+    this.getNotifications().forEach(notification => {
+      this.dismiss(notification, 0);
+    });
+  }
+
+  /**
+   * Merge default options with notification options
+   */
+  private mergeWithDefaultOptions(options = {}): any {
+    return Object.assign({}, this.options, options);
+  }
+
+  /**
+   * Generate a unique id for the notification
```

Comment:
`an* unique id`, yes I know, I'm an annoying person 😄 Sorry ^^

Reply:
You're absolutely right to check this, but not on this point ^^
https://www.wordreference.com/enfr/unique

Reply:
Ohh okay 😅

Reply:
Actually `a unique` is right. You use `an` for words that begin with a vowel sound. Not letter. `unique` starts with a `y` sound, so it is `a unique`.

Reply:
Oh that was Marie's comment, didn't see